Assets/Scripts/Cat Scripts/MovementController.cs

```diff
@@ -109,9 +127,39 @@ private void RotateCharacter()
             Quaternion targetRotation = Quaternion.LookRotation(moveDirection, Vector3.up);
 
             // smooth interpolate between the actual rotation to target rotation 
-            //we use .Slerp because the movement is smoother than other functions that are linear
+            //we use .Slerp because the movement is smoother than other linear functions 
             transform.rotation = Quaternion.Slerp(transform.rotation, targetRotation, rotationSpeed * Time.deltaTime);
         }
     }
 
+    private void ValidationJump()
+    {
+        // get a global -y component transforming the local direction to global with TransformDirection
+        Vector3 dwn = transform.TransformDirection(Vector3.down);
+
+        // create the raycast origin with a little desphase
+        Vector3 rayOrigin = transform.position + Vector3.up * 0.1f;
+
+        Debug.DrawRay(rayOrigin, dwn * raycastDistance, Color.yellow); 
+
+        // if the raycast hit with other collider in direction down can Jump
+        //the raycast gonna ignore the player layer
+        canJump = Physics.Raycast(rayOrigin, dwn, raycastDistance, gameObject.layer);
+
+        /*if(Physics.Raycast(rayOrigin, dwn, raycastDistance, gameObject.layer))
+        {
+            canJump = true;
+        }
+        else
+        {
+            canJump = false;
+        }*/
```

Comment:
Commented-out code was committed, which should be avoided unless it is clearly marked with an explanation for why it is retained. This can cause confusion and clutter in the codebase.